oklayoutinflator/src/main/java/tech/okcredit/layout_inflator/OkLayoutInflater.kt

```diff
@@ -18,11 +20,15 @@ import kotlinx.coroutines.*
  * 3. does not support setting a {@link LayoutInflater.Factory} nor {@link LayoutInflater.Factory2}. Here we are Adding support
  * 4. There is no way to cancel ongoing inflation. here we are exposing coroutine job cancel method
  */
-class OkLayoutInflater(context: Context) {
+class OkLayoutInflater(val context: Context) : DefaultLifecycleObserver {
 
     private val mInflater: LayoutInflater = BasicInflater(context)
-    private val coroutineContext = SupervisorJob()
-    private val scope = CoroutineScope(Dispatchers.Default + coroutineContext)
+    val job = SupervisorJob()
+    private val scope = CoroutineScope(Dispatchers.Default + job)
+
+    init {
+        context.getLifeCycleOwner()?.lifecycle?.addObserver(this)
```

Comment:
Since [addObserver](https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-main/lifecycle/lifecycle-common/src/main/java/androidx/lifecycle/Lifecycle.java?source=post_page---------------------------%2F%2F%2F%2F%2F%2F%2F&autodive=0%2F#66) enforces the main thread, the background thread cannot inflate the view call.

Reply:
I am using a low-end phone for testing and "OkLayoutInflater" a few milliseconds delay while inflating a view, that is a good make we are making view inflation thread-safe by default it is not. That delay is still there which means inflation works as expected ,adding an observer will not affect coroutines.

Reply:
@anjalsaneen If you can share some logs or screenshot for the above code review. how does this affect the view inflation?